File: API/Entities/OrderAggregate/Order.cs
using System.ComponentModel.DataAnnotations;

namespace API.Entities.OrderAggregate
{
    public class Order
    {
        public int Id { get; set; }
        public string BuyerId { get; set; }

        [Required] public ShippingAddress ShippingAddress { get; set; } = default!;

        public List<OrderItem> OrderItems { get; set; } = default!;
        public DateTime OrderDate { get; set; } = DateTime.Now.ToUniversalTime();
        public long SubTotal { get; set; }
        public long DeliveryFee { get; set; }
        public OrderStatus Status { get; set; } = OrderStatus.Pending;

        public long GetTotal()
        {
            return SubTotal + DeliveryFee;
        }
    }
}

File: client/src/app/models/order.ts
export interface Address {
    fullName: string;
    address1: string;
    address2: string;
    city: string;
    state: string;
    zip: string;
    country: string;
}

export interface OrderItem {
    productId: number;
    name: string;
    pictureUrl: string;
    quantity: number;
    price: number;
}

export interface OrderValue {
    saveAddress: boolean;
    shippingAddress: Address;
}

export interface Order {
    id: number;
    buyerId: string;
    address: Address;
    orderItems: OrderItem[];
    orderDate: Date;
    subTotal: number;
    orderStatus: string;
    deliveryFee: number;
    total: number;
}


